step-strikers-app/step-strikers-app/Classes/Armor.swift

```diff
@@ -0,0 +1,123 @@
+    
+    // Only Rogues and Bards can wear this armor
+    func checkIfSuited(potentialWearer: RPGCharacter) -> Bool{
+        return potentialWearer is Rogue || potentialWearer is Bard
+    }
+}
+
+struct padding: Armor {
+    let name = "Padding"
+    let armorClass = 11
+    var condition = 4
+    var useCount = 0
+    
+    // Only Rogues and Bards can wear this armor
+    func checkIfSuited(potentialWearer: RPGCharacter) -> Bool{
+        return potentialWearer is Rogue || potentialWearer is Bard
+    }
+}
+
+struct studdedLeather: Armor {
+    let name = "Studded Leather"
+    let armorClass = 12
+    var condition = 4
+    var useCount = 0
+    
+    // Only Rogues and Bards can wear this armor
+    func checkIfSuited(potentialWearer: RPGCharacter) -> Bool{
+        return potentialWearer is Rogue || potentialWearer is Bard
+    }
+}
+
+struct chainMail: Armor {
+    let name = "Chain Mail"
+    let armorClass = 16
+    var condition = 4
+    var useCount = 0
+    
+    // Only Rogues and Bards can wear this armor
+    func checkIfSuited(potentialWearer: RPGCharacter) -> Bool{
+        return potentialWearer is Fighter
+    }
+}
+
+struct shield: Armor {
+    var name = "Shield"
+    var armorClass = 2
+    var condition = 4
+    var useCount = 0
+    
+    // Only Rogues and Bards can wear this armor
+    func checkIfSuited(potentialWearer: RPGCharacter) -> Bool{
+        return potentialWearer is Fighter
+    }
+}
+
+struct noArmor: Armor {
+    var name = "No Armor"
+    var armorClass = 0
+    var condition = 4
+    var useCount = 0
+
+    func checkIfSuited(potentialWearer: RPGCharacter) -> Bool {
+        return true
+    }
+}
+
+// This function will return the modified armor class in case the wearer is ill-suited for their currArmor
+func calculateModifiedArmorClass(wearer: RPGCharacter) -> Int {
+    if(wearer.currArmor.checkIfSuited(potentialWearer: wearer)){
+        return wearer.currArmor.armorClass
+    } else {
+        return rollDie(quant: 1, sides: wearer.currArmor.armorClass)
+    }
+}
+
+func adjustCondition(armorUsed: inout Armor){
+    
+    if !(armorUsed is noArmor) {
+        armorUsed.useCount += 1
+        // checks if conditionBoundary contains the useCount, if so change condition
+        if conditionBoundary.contains(where: {$0.boundary == armorUsed.useCount}) {
+            // changingIndex holds the index in conditionBoundary with the new condition of the weapon
+            let changingIndex = conditionBoundary.firstIndex(where: {$0.boundary == armorUsed.useCount})
+            
+            // new Condition is the new condition of weaponUsed
+            let newCondition: String = conditionBoundary[changingIndex!].condition
+            
+            // Set the condition Int in weaponUsed to new Int associated with the newCondition
+            armorUsed.condition = conditionIntEquivalent[conditionIntEquivalent.firstIndex(where: {$0.condition == newCondition})!].intEquivalent
+        }
+    }
+    
+}
```

Comment:
Dang I don't understand this function at all! But it looks good to me if you tested it and everything works. What's conditionBoundary? This is just a question, don't block on this.